earth2studio/models/auto.py

```diff
@@ -14,85 +14,287 @@
+            fn = self._make_local_details(path)
+        kwargs["mode"] = mode
+
+        # call target filesystems open
+        self._mkcache()
+        if self.compression:
+            with self.fs._open(path, **kwargs) as f, open(fn, "wb") as f2:
+                if isinstance(f, AbstractBufferedFile):
+                    # want no type of caching if just downloading whole thing
+                    f.cache = BaseCache(0, f.cache.fetcher, f.size)
+                comp = (
+                    infer_compression(path)
+                    if self.compression == "infer"
+                    else self.compression
+                )
+                f = compr[comp](f, mode="rb")
+                data = True
+                while data:
+                    block = getattr(f, "blocksize", 5 * 2**20)
+                    data = f.read(block)
+                    f2.write(data)  # type: ignore
+        else:
+            if "callback" in kwargs:  # Patch here
+                self.fs.get_file(path, fn, callback=kwargs["callback"])
+            else:
+                self.fs.get_file(path, fn)
+        self.save_cache()
+        return self._open(path, mode)  # type: ignore
+
+
+class TqdmFormat(tqdm):
+    """Provides a `total_time` format parameter. Not used.
+    See: https://filesystem-spec.readthedocs.io/en/stable/api.html#fsspec.callbacks.TqdmCallback
+    """
+
+    @property
+    def format_dict(self) -> dict:
+        d = super().format_dict
+        return d
+
+
+class TqdmCallbackRelative(TqdmCallback):
+    """Simple extention of Tqdm callback to support progress bar on recurrive gets"""
+
+    def branched(self, path_1, path_2, **kwargs) -> Callback:  # type: ignore
+        """Child callback for recursive get"""
+        tqdm_kwargs = self._tqdm_kwargs
+        tqdm_kwargs["unit"] = "B"
+        tqdm_kwargs["unit_scale"] = True
+        tqdm_kwargs["unit_divisor"] = 1024
+        callback = TqdmCallback(
+            tqdm_kwargs=tqdm_kwargs,
+            tqdm_cls=self._tqdm_cls,
+        )
+        return callback
+
+
+class Package:
+    """A generic file system abstraction with local caching, uses Fsspec
+    WholeFileCacheFileSystem to manage files. Designed to be used for accessing remote
+    resources, particularly checkpoint files for pre-trained models. Presently supports
+    public folders on NGC, huggingface repos, s3 and any other built in file system
+    Fsspec supports.
 
     Parameters
     ----------
     root : str
         Root directory for file system
-    cache_dir : str, optional
-        Cache directory, if remote path files will be downloaded here. If none is
-        provided then the path set in LOCAL_CACHE enviroment variable will be used.
-        If that is empty the path "~/.cache/earth2studio" is used, by default None
+    fs : AbstractFileSystem | None, optional
+        The target filesystem to run under neath. If none is provided one will get
+        initialized based on root url, by default None
```

Comment:
Can we be more specific about what gets initialized.

Reply:
Updated!